### PR TITLE
Converted Addon to the V2 Format

### PR DESCRIPTION
**Pull request by cah-briangantzler**
Converted addon to a V2 format https://github.com/sinankeskin/ember-inputmask/issues/348
Updated the ember-modifier to use the new modify hook.

I updated some of the github ci paths because of the V2 format, but not sure how to test. Please look over there and see if there is anything else that would need fixed. Im not sure how you release to NPM, but that may also be slightly different due to the V2 format. Please read https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md These are the directions I followed. 

The last line does say
```
To publish, you will run npm publish in the addon directory.
```

### CODE REVIEW

**Comment by sinankeskin** [comment]
Sorry. I couldn't find free time to take a look. I definitely will on weekend.

File: addon/src/modifiers/inputmask.js [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
 
-  _initialize() {
-    this._setDefaults();
+    this.element = element;
 
-    const args = this.getArgs();
+    const args = this.getArgs(positional, named);
 
-    this._setInputMask(args);
+    new InputMask(args).mask(element);
```
Comment (sinankeskin) [inline addon/src/modifiers/inputmask.js:39]: Looks like you removed _setInputMask invoke here but method still exists. I believe we should revert this line back.

File: addon/src/modifiers/inputmask.js [hunk 2]
```diff
@@ -54,7 +64,7 @@ export default class InputmaskModifier extends Modifier {
     );
   }
 
-  _setInputMask(args) {
-    new Inputmask(args).mask(this.element);
+  _setInputMask(element, args) {
```
Comment (sinankeskin) [inline addon/src/modifiers/inputmask.js:67]: We should revert this back with just args parameter IMO.

File: addon/src/modifiers/inputmask.js [hunk 3]
```diff
@@ -1,39 +1,49 @@
 /* globals Inputmask */
 import Modifier from 'ember-modifier';
 import { getOwner } from '@ember/application';
+import InputMask from "inputmask";
```
Comment (sinankeskin) [inline addon/src/modifiers/inputmask.js:4]: Could you please run `yarn:lint:fix` inside the projects?

File: test-app/tests/integration/components/inputmask-test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    this.testClass = new TestClass();
+
+    this.value = undefined;
+
+    this.onUpdate = (value) => {
+      this.value = value;
+    };
+
+    debugger;
```
Comment (sinankeskin) [inline test-app/tests/integration/components/inputmask-test.js:69]: You forgot 3 `debugger` invokes in this file.

**Review by sinankeskin — CHANGES_REQUESTED** [review]
Quite big change but overall looks good and works good. Just minor things. Thank you @cah-briangantzler Appreciated.

File: test-app/tests/integration/components/inputmask-test.js (hunk 4 above)
Comment (cah-briangantzler) [reply test-app/tests/integration/components/inputmask-test.js:69]: thats embarrassing :)

File: addon/src/modifiers/inputmask.js (hunk 3 above)
Comment (cah-briangantzler) [reply addon/src/modifiers/inputmask.js:4]: In all my personal projects I use lint-staged to always run lint and prettier when I commit, so it just happens. I always forget this when doing external projects.

File: addon/src/modifiers/inputmask.js (hunk 1 above)
Comment (cah-briangantzler) [reply addon/src/modifiers/inputmask.js:39]: I meant to delete the method. Its a one line method that is invoke in only one place now. Also no longer needs to return.

Its your call. I think deleting the method is the way to go. You want me to add back the invoke?

File: addon/src/modifiers/inputmask.js (hunk 2 above)
Comment (cah-briangantzler) [reply addon/src/modifiers/inputmask.js:67]: see previous comment

File: addon/src/modifiers/inputmask.js (hunk 1 above)
Comment (sinankeskin) [reply addon/src/modifiers/inputmask.js:39]: Nope I meant if you want to keep it we should revert this line back. If you want to remove totally fine.

**Comment by sinankeskin** [comment]
We need to change `ci.yml` according to latest changes @cah-briangantzler Tests are failing. Actually doesn't event start.

**Comment by cah-briangantzler** [comment]
Yea, I didnt have any real way of simulating the ci locally. Tests were suppose to be able to run by adding the working dir. I thinking because of the yarn workspaces there needs to be a yarn somewhere else. Not that I have a log will try a few things, but may take a couple of attempts

**Comment by sinankeskin** [comment]
I don't want give you too much trouble. We could disable for now if you'd like. I can merge and publish because this state is far from I can image. All looks and works great.

**Comment by cah-briangantzler** [comment]
Let me have a couple of attempts. You are using `npm ci` which requires a lock file, there isnt a lock file at the root because I used `yarn`. Do you have a preference? I usually use yarm but can switch it over to npm easy enough

**Comment by sinankeskin** [comment]
No I don't have any preference. I used `npm` just because it was default. Actually I use `yarn` all the time with my job projects.

**Comment by sinankeskin** [comment]
Error: ENOENT: no such file or directory, lstat '/home/runner/work/ember-inputmask/ember-inputmask/addon/./dist/_app_/components/inputmask.js'

Either we need to run `yarn prepublishOnly` first or we need to send `dist` folders too I guess.

**Comment by cah-briangantzler** [comment]
We need to insert a 
```
yarn build
```
somewhere Im thinking

**Comment by cah-briangantzler** [comment]
Updated and added the `prepare` script. I think that might be it.  Using https://github.com/kaliber5/ember-stargate/blob/master/.github/workflows/ci.yml as an example. I added a working directory and run the tests directly. He has a `test` script at the root that redirects to the `test` in the package. He (or other examples I looked at) do not check in dist. I think the missing piece was https://github.com/kaliber5/ember-stargate/blob/master/packages/ember-stargate/package.json#L29

You will also note this, https://github.com/bgantzler/ember-inputmask/blob/addon-v2-format/addon/package.json#L19 which I believe it what creates the dist for when it is published to npm. That was gotten from here https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md#part-4-convert-addon-to-v2 step 15

**Comment by sinankeskin** [comment]
Hey @cah-briangantzler added you as a collaborator to this repo. You can run the test whenever you want now.

**Comment by cah-briangantzler** [comment]
WooHoo.

Hope you are able to make the needed changes to release it. Ember-stargate uses release-it and here is some of what he has in the package.json https://github.com/kaliber5/ember-stargate/blob/master/package.json#L31. Maybe that will help as an example

**Comment by sinankeskin** [comment]
> 

Awesome job. I don't wanna use another package. I'll try to release with just npm. I'll try to figure it out and let you know.

Thank you @cah-briangantzler Appreciated.

**Comment by cah-briangantzler** [comment]
Re-read the first message, it has a link to the docs I used and steps to release

**Comment by sinankeskin** [comment]
Got it. Thank you. Merging.  🎉

**Comment by cah-briangantzler** [comment]
https://github.com/kaliber5/ember-stargate/blob/master/package.json#L51 seems we may need the version number at the root package.

Attempting to link to packge using #main is not working.

**Comment by sinankeskin** [comment]
Makes sense but I just released. 🎉  Next time I'll update. Thank you again @cah-briangantzler

**Pull request merged**